crates/js-component-bindgen/src/lib.rs

```diff
@@ -172,3 +172,35 @@ pub fn dealias(resolve: &Resolve, mut id: TypeId) -> TypeId {
         }
     }
 }
+
+/// Check if an item (usually some form of [`WorldItem`]) should be allowed through the feature gate
+/// of a given package.
+fn feature_gate_allowed(
+    resolve: &Resolve,
+    package: &Package,
+    stability: &Stability,
+) -> Result<bool> {
+    Ok(match stability {
+        Stability::Unknown => true,
+        Stability::Stable { since, .. } => {
+            let Some(package_version) = package.name.version.as_ref() else {
+                // If the package version is missing (we're likely dealing with an unresolved package)
+                // and we can't really check much.
+                return Ok(true);
+            };
+
+            ensure!(
+                package_version >= since,
+                "feature gates cannot refer to unreleased (future) package versions"
+            );
```

Comment:
For this error, in case it happens to someone for whatever reason, can we include the name of the function that it applies to, by adding a `debug_name` or otherwise to this function?

Reply:
Absolutely -- that would improve ease of debugging -- I'll add the versions as well.

- [ ] Add item being checked to stability fn, show versions that conflict